src/Parser/Mage/Fire/Modules/Features/HeatingUp.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import Wrapper from 'common/Wrapper';
+import { formatMilliseconds, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+import AbilityTracker from 'Parser/Core/Modules/AbilityTracker';
+import EnemyInstances, { encodeTargetString } from 'Parser/Core/Modules/EnemyInstances';
+
+const debug = false;
+
+const GUARANTEE_CRIT_SPELLS = [
+  SPELLS.FIRE_BLAST.id,
+  SPELLS.PHOENIXS_FLAMES.id,
+];
+
+class HeatingUp extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    abilityTracker: AbilityTracker,
+    enemies: EnemyInstances,
+  };
+
+  fireBlastWithoutHeatingUp = 0;
+  phoenixFlamesWithoutHeatingUp = 0;
+  fireBlastWithHotStreak = 0;
+  phoenixFlamesWithHotStreak = 0;
+  targetId = 0;
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.PHOENIXS_FLAMES.id) {
+      return;
+    }
+    this.targetId = encodeTargetString(event.targetID, event.targetInstance);
+  }
+
+  on_byPlayer_damage(event) {
+    const spellId = event.ability.guid;
+    const damageTarget = encodeTargetString(event.targetID, event.targetInstance);
+    if (!GUARANTEE_CRIT_SPELLS.includes(spellId) || (spellId === SPELLS.PHOENIXS_FLAMES.id && (this.targetId !== damageTarget))) {
+      return;
+    }
+    const hasHeatingUp = this.combatants.selected.hasBuff(SPELLS.HEATING_UP.id);
+    const hasHotStreak = this.combatants.selected.hasBuff(SPELLS.HOT_STREAK.id);
+
+    switch (spellId) {
+      case SPELLS.FIRE_BLAST.id:
+        if (hasHotStreak) {
+          this.fireBlastWithHotStreak += 1;
+          debug && console.log("Fire Blast with Hot Streak @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        if (!hasHeatingUp && !hasHotStreak) {
+          this.fireBlastWithoutHeatingUp += 1;
+          debug && console.log("Fire Blast without Heating Up @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        break;
+      case SPELLS.PHOENIXS_FLAMES.id:
+        if (hasHotStreak) {
+          this.phoenixFlamesWithHotStreak += 1;
+          debug && console.log("Phoenix Flames with Hot Streak @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+        }
+        if (!hasHeatingUp && !hasHotStreak) {
```

Comment:
Could be `else if (!hasHeatingUp) {` because the `else` ensures you don't have hot streak